File: backend-springboot/src/main/java/com/monitor/controller/DashboardController.java
package com.monitor.controller;

import com.monitor.service.DashboardService;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

/**
 * Dashboard 控制器
 */
@RestController
@RequestMapping("/dashboard")
@RequiredArgsConstructor
public class DashboardController {
    
    private final DashboardService dashboardService;
    
    @Value("${default-project-id:001}")
    private String defaultProjectId;
    
    @GetMapping("/overview")
    public ResponseEntity<Map<String, Object>> overview() {
        Map<String, Object> data = dashboardService.getOverview(defaultProjectId);
        return ResponseEntity.ok(Map.of("success", true, "data", data));
    }
}

File: backend-midway/src/controller/dashboard.controller.ts
import { Controller, Get, Inject, Config } from '@midwayjs/core';
import { ApiTags, ApiOperation } from '@midwayjs/swagger';
import { DashboardService } from '../service/dashboard.service';

@Controller('/dashboard')
@ApiTags(['dashboard'])
export class DashboardController {
  @Inject()
  dashboardService: DashboardService;

  @Config('defaultProjectId')
  defaultProjectId: string;

  @Get('/overview')
  @ApiOperation({ summary: '获取仪表盘概览数据' })
  async overview() {
    const data = await this.dashboardService.getOverview(this.defaultProjectId);
    return { success: true, data };
  }
}

